miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt

```diff
@@ -98,8 +112,10 @@ internal class MiniAppDownloader(
                     val response = apiClient.downloadFile(file)
                     storage.saveFile(file, baseSavePath, response.byteStream())
                 }
-                withContext(coroutineDispatcher) {
-                    launch(Job()) { storage.removeOutdatedVersionApp(appId, versionId) }
+                if (!apiClient.isPreviewMode) {
```

Comment:
Nice, so preview mode can have multiple versions downloaded, but published mode can only have one version downloaded?

Reply:
That's right and I also change the target of hash validation from miniapp id into version id to keep cache of multiple versions in preview mode.